pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -372,50 +338,15 @@ func (c *NPRecommendationController) updateResult(npReco *crdv1alpha1.NetworkPol
 			},
 		)
 	}
-
-	if npReco.Status.RecommendedNP == nil {
-		err := deleteSparkApplicationIfExists(c.kubeClient, npReco.Namespace, npReco.Status.SparkApplication)
-		if err != nil {
-			return fmt.Errorf("fail to delete Spark Application: %s, error: %v", npReco.Status.SparkApplication, err)
-		}
-		var recommendedNetworkPolicy *crdv1alpha1.RecommendedNetworkPolicy
-		// Check if RecommendedNetworkPolicy CR is created
-		rnps, err := c.recommendedNPInformer.GetIndexer().ByIndex(idIndex, npReco.Status.SparkApplication)
-		if err == nil && len(rnps) > 0 {
-			// Use the existing RecommendedNetworkPolicy CR
-			recommendedNetworkPolicy = rnps[0].(*crdv1alpha1.RecommendedNetworkPolicy)
-			if len(rnps) > 1 {
-				klog.V(4).InfoS("More than 1 RecommendedNetworkPolicy", "id", npReco.Status.SparkApplication)
-			}
-		} else {
-			if err != nil {
-				klog.V(4).InfoS("Failed to find RecommendedNetworkPolicy", "id", npReco.Status.SparkApplication)
-			}
-			// Get result from database
-			if c.clickhouseConnect == nil {
-				c.clickhouseConnect, err = setupClickHouseConnection(c.kubeClient, npReco.Namespace)
-				if err != nil {
-					return err
-				}
-			}
-			recommendedNetworkPolicy, err = getPolicyRecommendationResult(c.clickhouseConnect, npReco.Status.SparkApplication)
-			if err != nil {
-				return err
-			}
-			// Create RecommendedNetworkPolicy CR
-			recommendedNetworkPolicy, err = c.crdClient.CrdV1alpha1().RecommendedNetworkPolicies(npReco.Namespace).Create(context.TODO(), recommendedNetworkPolicy, metav1.CreateOptions{})
-			if err != nil {
-				return err
-			}
-			klog.V(2).InfoS("Created RecommendedNetworkPolicy", "RecommendedNetworkPolicy", recommendedNetworkPolicy.Name, "NetworkRecommendationPolicy", npReco.Name)
-		}
-		return c.updateNPRecommendationStatus(npReco, crdv1alpha1.NetworkPolicyRecommendationStatus{
-			State:         crdv1alpha1.NPRecommendationStateCompleted,
-			RecommendedNP: recommendedNetworkPolicy,
-			EndTime:       metav1.NewTime(time.Now()),
-		})
+	// Delete related SparkApplication CR
+	err := deleteSparkApplicationIfExists(c.kubeClient, npReco.Namespace, npReco.Status.SparkApplication)
```

Comment:
This is not related to your PR, so please take this comment as optional or a hint for a different PR.
It's not very efficient that we list all spark applications before finding the one to delete. In a large scale env we can have thousands of them. If deletion fails because the app does not exists, can we just handle the failure (and ignore it)?

Reply:
That makes sense from the efficiency perspective. Updated, thanks!